source/adapters/level_zero/adapter.cpp

```diff
@@ -12,14 +12,27 @@
 #include "ur_level_zero.hpp"
 #include <iomanip>
 
+// As windows order of unloading dlls is reversed from linux, windows will call
+// umfTearDown before it could release umf objects in level_zero, so we call
+// umfInit on urAdapterGet and umfAdapterTearDown to enforce the teardown of umf
+// after umf objects are destructed.
+#if defined(_WIN32)
+#include <umf.h>
+#endif
+
 // Due to multiple DLLMain definitions with SYCL, Global Adapter is init at
 // variable creation.
 #if defined(_WIN32)
 ur_adapter_handle_t_ *GlobalAdapter = new ur_adapter_handle_t_();
 #else
 ur_adapter_handle_t_ *GlobalAdapter;
 #endif
-
+// This is a temporary workaround on windows, where UR adapter is teardowned
+// before the UR loader, which will result in access violation when we use print
+// function as the overrided print function was already released with the UR
+// adapter.
+// TODO: Change adapters to use a common sink class in the loader instead of
+// using thier own sink class that inherit from logger::Sink.
```

Comment:
This is not ideal, but it should work for now to solve a release problem regarding the logger. @igchor @nrspruit If this is okay, I will merge this asap.

Reply:
Yeah, I think that's fine for now. Eventually, we could do a similar thing like here: https://github.com/oneapi-src/unified-runtime/pull/2000. Just keep a global instance of shared_ptr<> to the logger (here https://github.com/oneapi-src/unified-runtime/blob/b766009add8dc41ad03e2a63be9f6ed40eea4c55/source/common/logger/ur_logger.hpp#L24) and every time get_logger is called, we would return shared_ptr instead of a reference.